pkg/utils/managedresources/registry.go

```diff
@@ -92,6 +109,25 @@ func (r *Registry) Add(objs ...client.Object) error {
 			return err
 		}
 
+		// We use a copy of the upstream package as a workaround
+		// to incosistent ordering in yaml.v2 when encoding maps
+		// Can be removed once k8s.io/apimachinery/pkg/runtime/serializer/json migrates to yaml.v3
+		// or the issue is resolved upstream in yaml.v2
+		if r.isSerializerYAML {
+			// we do not handle arrays since the iterated variables
+			// are of type client.Object
+			anyObj := map[string]any{}
```

Comment:
```suggestion
			var yamlObj interace{}
```

I'm wondering why we shouldn't use `interace{}` here and leave out the comment? There is probably no special handling/logic needed from our side. Just let the yaml lib handle it? Or is there a special reason?

Reply:
Basically https://go.dev/play/p/bzufE8P1z9Z

Then the yaml library panics with `reflect: call of reflect.Value.Type on zero Value`

Reply:
Try https://go.dev/play/p/HQ0OqCuJ2PG

```
forkedyaml.Unmarshal(serializationYAML, &anyObj)
```

Reply:
You are absolutely right, thanks!